DLSv2/Core/ManagedVehicle.cs

```diff
@@ -324,22 +324,22 @@ public void RegisterInputs()
                     ControlsManager.PlayInputSound();
 
                     // Clears light modes
-                    foreach (string key in StandaloneLightModes.Keys.ToList())
-                        StandaloneLightModes[key] = false;
+                    foreach (var mode in LightModes.Values)
+                        mode.Enabled = false;
 
                     // Clears light control groups
-                    foreach (string key in LightControlGroups.Keys.ToList())
-                        LightControlGroups[key] = (false, 0);
+                    foreach (var cG in LightControlGroups.Values.ToList())
+                        cG.Disable();
 
                     // Updates lights
-                    LightController.Update(this);
+                    UpdateLights();
 
                     // Clears audio control groups
-                    foreach (string key in AudioControlGroups.Keys.ToList())
-                        AudioControlGroups[key] = (false, 0);
+                    foreach (var cG in AudioControlGroups.Values.ToList())
```

Comment:
Do these all need to have `.ToList()`? I think the `foreach` should work fine directly on a `Values` collection.

Reply:
IIRC this is in so I can set the value within the collection while looping.

Reply:
If not changing the keys and only changing the values, it should be safe to loop over the keys directly and set the corresponding values.